### PR TITLE
Kwargs parsing

### PR DESCRIPTION
**Pull request by danielytan**
Added support for parsing keyword arguments:
- collects kwargs into a `Map<string, Expr<Location>>` for use in `call_expr` and `method-call`
- enforces that positional arguments occur before keyword arguments
- no keywords can be repeated

### CODE REVIEW

File: ast.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export type Scope<A> =
 
 export type Parameter = { name: string; type: Type; value?: Literal };
 
+export type Argument<A> = { a?: A; name?: string; value: Expr<A> }
```
Comment (DavidLiu0304) [inline ast.ts]: I don't think argument type is used anywhere. You should remove this

File: parser.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ import {
   Location,
 } from "./ast";
 
-import { NUM, BOOL, NONE, CLASS, isTagged, STRING, LIST } from "./utils";
+import { NUM, BOOL, NONE, CLASS, isTagged, STRING, LIST, stringify } from "./utils";
```
Comment (DavidLiu0304) [inline parser.ts]: Don't think stringify is used

File: ast.ts (hunk 1 above)
Comment (danielytan) [reply ast.ts]: Good point; we separate keyword arguments and arguments explicitly now so we don't need it anymore!

File: parser.ts (hunk 2 above)
Comment (danielytan) [reply parser.ts]: oops you're right; I think I accidentally added it after mistyping string.

**Review by DavidLiu0304 — APPROVED** [review]
Look good to me

**Pull request merged**